### PR TITLE
update watchlist when set watched percentage is reached

### PR DESCRIPTION
**Pull request by SwagOtaku**
as suggested in discord

> Kodi can we have an option to auto update MAL instead of asking me Everytime I finish an episode?

### CODE REVIEW

**Comment by Prometheusx-git** [comment]
Does it also work with myanimelist or only with kitsu? 
Until now there was no update on myanimelist even after asking for it.

**Comment by SwagOtaku** [comment]
@Prometheusx-git watchlist update has been working for both mal and kitsu since we introduced it

File: default.py [hunk 1]
```diff
@@ -58,6 +58,14 @@ def sortResultsByRes(fetched_urls):
                   utils.parse_resolution_of_source(x[0]),
                   reverse=True)
 
+#Will be called at handle_player
+def on_percent():
+    percent_bool = True if 'true' in control.getSetting('watchlist.percentbool') else False
```
Comment (DxCx) [inline default.py]: You can register callback or pass None depends on setting instead of always pass the callback and check inside :)

**Comment by SwagOtaku** [comment]
whoops i messed up

**Comment by SwagOtaku** [comment]
@DxCx ready for re-review

File: default.py [hunk 2]
```diff
@@ -207,7 +215,7 @@ def PLAY(payload, params):
     })
 
     __set_last_watched(anime_url, is_dubbed, name, image)
-    control.play_source(s.get_video_link(), watchlist_update(episode, kitsu_id), on_stopped)
+    control.play_source(s.get_video_link(), watchlist_update(episode, kitsu_id), on_stopped, on_percent)
```
Comment (DxCx) [inline default.py]: you can pass here None if not control.getSetting('watchlist.percentbool'),
then callback can just `return int(control.getSetting('watchlist.percent'))`

**Comment by DxCx** [comment]
Awsome

**Pull request merged**